R/MsBackend.R

```diff
@@ -127,6 +127,10 @@
 #' - `$`, `$<-`: access or set/add a single spectrum variable (column) in the
 #'   backend.
 #'
+#' - `[[`, `[[<-`: access or set/add a single spectrum variable (column) in the
+#'   backend. The default implementation uses `$`, thus these methods don't have
```

Comment:
Shouldn't it be the other way around? Meaning: The default implementation of `$` uses `[[`.

`?"$"`
> Both ‘[[’ and ‘$’ select a single element of the list.  The main
     difference is that ‘$’ does not allow computed indices, whereas
     ‘[[’ does.  ‘x$name’ is equivalent to ‘x[["name", exact =
     FALSE]]’.  Also, the partial matching behavior of ‘[[’ can be
     controlled using the ‘exact’ argument.

Reply:
Yes, that would make more sense - only we required `$` and `$<-` for all backends that far, so all backends have these methods implemented, but not `[[` and `[[<-`. With the current implementation (`[[` is based on `$`) all backends gain this method without breaking anything.

Reply:
@sgibb - is it OK to leave it like that, even if `[[` and `[[<-` should be those that `$` and `$<-` reply on. Implementation wise, it doesn't matter, as far as I see. It's historical in this case.